src/libguac/protocol.c

```diff
@@ -961,6 +1004,26 @@ int guac_protocol_send_rect(guac_socket* socket,
 
 }
 
+int guac_protocol_send_required(guac_socket* socket, const char** required) {
+    
+    int ret_val;
+    
+    guac_socket_instruction_begin(socket);
+    
+    if (guac_socket_write_string(socket, "8.required"))
+        return -1;
+
+    if (guac_socket_write_array(socket, required))
+        return -1;
+
+    ret_val = guac_socket_write_string(socket, ";");
```

Comment:
FYI, given the definition of the return value of this function, this is equivalent to:

```
ret_val =
       guac_socket_write_string(socket, "8.required")
    || guac_socket_write_array(socket, required)
    || guac_socket_write_string(socket, ";");
```

Reply:
I like that a lot better!